router/core/timout_transport.go

```diff
@@ -0,0 +1,44 @@
+package core
+
+import (
+	"context"
+	"go.uber.org/zap"
+	"net/http"
+)
+
+type TimeoutTransport struct {
+	defaultTransport http.RoundTripper
+	logger           *zap.Logger
+	subgraphTrippers map[string]*http.Transport
+	opts             *SubgraphTransportOptions
+}
+
+func NewTimeoutTransport(transportOpts *SubgraphTransportOptions, roundTripper http.RoundTripper, logger *zap.Logger, proxy ProxyFunc) *TimeoutTransport {
+	tt := &TimeoutTransport{
+		defaultTransport: roundTripper,
+		logger:           logger,
+		subgraphTrippers: map[string]*http.Transport{},
+		opts:             transportOpts,
+	}
+
+	for subgraph, subgraphOpts := range transportOpts.SubgraphMap {
+		tt.subgraphTrippers[subgraph] = newHTTPTransport(*subgraphOpts, proxy)
+	}
+
+	return tt
+}
+
+func (tt *TimeoutTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	rq := getRequestContext(req.Context())
+	subgraph := rq.ActiveSubgraph(req)
+	if subgraph != nil && subgraph.Name != "" && tt.subgraphTrippers[subgraph.Name] != nil {
+		timeout := tt.opts.SubgraphMap[subgraph.Name].RequestTimeout
```

Comment:
Why do we have to enforce a timeout here when we have set it globally on the transport?

Reply:
The idea is that you can have a global timeout, but a user could set a different request timeout for the specific subgraph. Given that we set the timeout per client, we don't really have an ability otherwise to inject the request timeout, from what I had seen

Reply:
I see, thanks for clarifying.